lightning/src/ln/channelmanager.rs

```diff
@@ -1333,6 +1339,39 @@ impl<ChanSigner: ChannelKeys, M: Deref, T: Deref, K: Deref, F: Deref> ChannelMan
 		})
 	}
 
+	/// Generates a signed node_announcement from the given arguments and creates a
+	/// BroadcastNodeAnnouncement event.
+	///
+	/// RGB is a node "color" and alias ia a printable human-readable string to describe this node
+	/// to humans. They carry no in-protocol meaning.
+	///
+	/// addresses represent the set (possibly empty) of socket addresses on which this node accepts
+	/// incoming connections. These will be broadcast to the network, publicly tying these
+	/// addresses together. If you wish to preserve user privacy, addresses should likely contain
+	/// only Tor Onion addresses.
+	pub fn broadcast_node_announcement(&self, rgb: [u8; 3], alias: [u8; 32], addresses: msgs::NetAddressSet) {
+		let _ = self.total_consistency_lock.read().unwrap();
+
+		let announcement = msgs::UnsignedNodeAnnouncement {
+			features: NodeFeatures::supported(),
+			timestamp: self.last_node_announcement_serial.fetch_add(1, Ordering::AcqRel) as u32,
+			node_id: self.get_our_node_id(),
+			rgb, alias,
+			addresses: addresses.into_vec(),
+			excess_address_data: Vec::new(),
```

Comment:
Since this is our node announcement, we'll never have any excess address data, right? this is just for other nodes that may randomly have excess address data? is this a common problem...?

Reply:
That is correct. And, indeed, it is *not* a common thing. We have to have it as otherwise the signatures of things we relay will fail, but, in general, we anticipate almost never having anything in there, or if we do, a very small thing.